src/tpm2_getmanufec.c

```diff
@@ -271,18 +272,24 @@ unsigned char *HashEKPublicKey(void)
     fp = fopen(outputFile, "rb");
     if (fp == NULL)
         printf("File Open Error\n");
-    else 
+    else
     {
         fseek(fp, 0x66, 0);
-        fread(EKpubKey, 1, 256, fp);
+        size_t read = fread(EKpubKey, 1, 256, fp);
+        if (read != 256)
+        {
+            printf ("Could not read whole file.");
```

Comment:
Sure, but the old code didn't check this, but will fix this issue since were here.